kong/plugins/statsd/schema.lua

```diff
@@ -6,19 +6,130 @@ local metrics = {
   "response_size",
   "unique_users",
   "request_per_user",
-  "upstream_latency"
+  "upstream_latency",
+  "kong_latency",
+  "status_count_per_user"
 }
 
+local stat_types = {
+  "gauge",
+  "timer",
+  "counter",
+  "histogram",
+  "meter",
+  "set"
+}
+
+local consumer_identifiers = {
+  "consumer_id",
+  "custom_id",
+  "username"
+}
+
+local default_metrics = {
+  {
+    name = "request_count",
+    stat_type = "counter",
+    sample_rate = 1
+  },
+  {
+    name = "latency",
+    stat_type = "timer"
+  },
+  {
+    name = "request_size",
+    stat_type = "timer"
+  },
+  {
+    name = "status_count",
+    stat_type = "counter",
+    sample_rate = 1
+  },
+  {
+    name = "response_size",
+    stat_type = "timer"
+  },
+  {
+    name = "unique_users",
+    stat_type = "set",
+    consumer_identifier = "custom_id"
+  },
+  {
+    name = "request_per_user",
+    stat_type = "counter",
+    sample_rate = 1,
+    consumer_identifier = "custom_id"
+  },
+  {
+    name = "upstream_latency",
+    stat_type = "timer"
+  },
+  {
+    name = "kong_latency",
+    stat_type = "timer"
+  },
+  {
+    name = "status_count_per_user",
+    stat_type = "counter",
+    sample_rate = 1,
+    consumer_identifier = "custom_id"
+  }
+}
+
+local function check_value(table, element)
+  for _, value in pairs(table) do
+    if value == element then
+      return true
+    end
+  end
+  return false
+end
```

Comment:
Yes I ll update the logic, though it will hardly make any difference as `consumer_identifiers` is a fixed length table.